Move packaging config to pyproject.toml

kimbyungnam opened the pull request:
* move packaging metadata to [PEP 621](https://peps.python.org/pep-0621/) in pyproject.toml
* drop setup.py/setup.cfg

## Testing

* pip install .
* python -m build

kimbyungnam committed:
Move packaging config to pyproject.toml

adamreeve commented:
Thanks for this @kimbyungnam. There are also a few places that use or reference `setup.py` that will need to be updated:
```
$ git grep 'setup\.py'
.github/workflows/ci-cd.yml:          python setup.py sdist
README.rst:    python setup.py install
docs/quickstart.rst:    python setup.py install
```

kimbyungnam committed:
Remove deprecated license classifiers (PEP 639)

Trove license classifiers are now deprecated and replaced by SPDX-based
license metadata as defined in PEP 639

kimbyungnam committed:
Update docs and publish workflow for pyproject build

kimbyungnam commented:
@adamreeve Thanks for the review!

I’ve updated the places that were still referencing `setup.py`:

- In `.github/workflows/ci-cd.yml`, the packaging step now uses `python -m build` driven by `pyproject.toml` instead of `setup.py`.
- In `README.rst` and `docs/quickstart.rst`, the installation instructions now use `python -m pip install .`.

While I was there, I also:

- Switched `docs/conf.py` to read the version from `pyproject.toml`.
- Removed the deprecated Trove license classifiers in favour of SPDX-based license metadata as described in PEP 639.

Please let me know if you’d like any of this done differently or split out into separate commits.

adamreeve commented:
:+1: good catch fixing the version parsing in the docs too. Thanks for this!
